### PR TITLE
Ban pagination

### PR DESCRIPTION
**Pull request by BobDotCom**
## Summary

Implements ban paginations

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [x] If `type: ignore` comments were used, a comment is also left explaining why
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [x] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

**Review by krittick — DISMISSED** [review]

File: discord/iterators.py [hunk 1]
```diff
@@ -678,14 +678,74 @@ def create_member(self, data):
...
-        guild: Guild,
-        limit: Optional[int],
-        joined: bool,
-        private: bool,
+            self,
+            channel_id: int,
+            guild: Guild,
+            limit: Optional[int],
+            joined: bool,
+            private: bool,
```
Comment (Middledot) [inline discord/iterators.py:748]:
```suggestion
        self,
        channel_id: int,
        guild: Guild,
        limit: Optional[int],
        joined: bool,
        private: bool,
```

File: discord/http.py [hunk 2]
```diff
@@ -1342,8 +1342,23 @@ def create_from_template(self, code: str, name: str, region: str, icon: Optional
...
         return self.request(Route("POST", "/guilds/templates/{code}", code=code), json=payload)
 
-    def get_bans(self, guild_id: Snowflake) -> Response[List[guild.Ban]]:
-        return self.request(Route("GET", "/guilds/{guild_id}/bans", guild_id=guild_id))
+    def get_bans(
+            self,
+            guild_id: Snowflake,
+            limit: Optional[int] = None,
+            before: Optional[Snowflake] = None,
+            after: Optional[Snowflake] = None,
```
Comment (Middledot) [inline discord/http.py:1350]:
```suggestion
        self,
        guild_id: Snowflake,
        limit: Optional[int] = None,
        before: Optional[Snowflake] = None,
        after: Optional[Snowflake] = None,
```

File: discord/iterators.py (hunk 1 above)
Comment (BobDotCom) [reply discord/iterators.py:748]: [PEP8](https://peps.python.org/pep-0008/#indentation) states "Add 4 spaces (an extra level of indentation) to distinguish arguments from the rest."

File: discord/http.py (hunk 2 above)
Comment (BobDotCom) [reply discord/http.py:1350]: [PEP8](https://peps.python.org/pep-0008/#indentation) states "Add 4 spaces (an extra level of indentation) to distinguish arguments from the rest."

**Review by Middledot — APPROVED** [review]

**Review by Dorukyum — APPROVED** [review]

**Review by krittick — APPROVED** [review]

**Pull request merged**